receiver/prometheusreceiver/config.go

```diff
@@ -37,7 +35,7 @@ type Config struct {
 	// ReportExtraScrapeMetrics - enables reporting of additional metrics for Prometheus client like scrape_body_size_bytes
 	ReportExtraScrapeMetrics bool `mapstructure:"report_extra_scrape_metrics"`
 
-	TargetAllocator *TargetAllocator `mapstructure:"target_allocator"`
+	TargetAllocator *targetallocator.Config `mapstructure:"target_allocator"`
```

Comment:
Sorry for the delay in review.

If you make field `TargetAllocator` of type `targetallocator.Config` where `targetallocator` is an internal package, you effectively prevent users from programmatically create this configuration. This can result in issues like this one for Kafka exporter: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33180.

I propose to store the configuration structs in a public package.

Reply:
I can't just move the config, as that would create a circular dependency between the directories.  Instead, i've moved the `targetallocator` package out of `internal`.  This exposes `Manager` and `NewManager` as part of the public API.